sdk/core/azure-core-amqp/src/amqp/message_sender.cpp

```diff
@@ -37,6 +37,24 @@ namespace Azure { namespace Core { namespace Amqp { namespace _detail {
 }}}} // namespace Azure::Core::Amqp::_detail
 
 namespace Azure { namespace Core { namespace Amqp { namespace _internal {
+  std::ostream& operator<<(std::ostream& stream, SenderSettleMode const& settleMode)
+  {
+    switch (settleMode)
+    {
+      case SenderSettleMode::Settled:
+        stream << "Settled";
+        break;
+      case SenderSettleMode::Unsettled:
+        stream << "Unsettled";
+        break;
+      case SenderSettleMode::Mixed:
+        stream << "Mixed";
+        break;
+      default:
+        throw std::runtime_error("Unknown message sender settle state.");
```

Comment:
Given that these operators are mostly used for logs, do you think it would be better to print "UNKNOWN" instead of throwing?

Let's say we added a new value, and did not update this operator. Tested with the log message never occurring, or with logs disabled. Shipped it, and it breaks for customer with logs enabled.

This would apply to all similar `operator <<`, if they are throwing as well.

Reply:
Better solution: Remove the default: case entirely.

If a new value is added, the compiler will complain that not all possible branches are taken.

Reply:
Maybe, but would they go out of their way? Technically, they don't have to - it is a valid C++ code to not have all values covered. You'll just get an empty string, compiler may think this is exactly what you wanted.
If at least one in our CI does that, I guess we're good.